File: dice/typescript/sol/src/apigear_game/random.test.ts
import { Random } from "./random";

test("Random.generate", () => {
  const rand = new Random();
  for (var i = 0; i < 10; i++) {
    const v = rand.generate(1, 6);
    expect(v).toBeGreaterThanOrEqual(1);
    expect(v).toBeLessThanOrEqual(6);
  }
});

File: dice/cpp14/ini/apigear_game/random.test.cpp
/**
NO TITLE
Copyright (C) 2020 ApiGear UG

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/
#include "catch2/catch.hpp"
#include "random.h"

TEST_CASE("Testing Random", "[Random]"){
    Random testRandom;
    // setup your test
    SECTION("Test operation generate") {
        // Do implement test here
        testRandom.generate(0,0);
    }
}
